bigint/src/lib.rs

```diff
@@ -333,13 +336,13 @@ fn from_inexact_bitwise_digits_le(v: &[u8], bits: usize) -> BigUint {
     let mut d = 0;
     let mut dbits = 0;
     for &c in v {
-        d |= (c as DoubleBigDigit) << dbits;
+        d |= (c as BigDigit) << dbits;
         dbits += bits;
+
         if dbits >= big_digit::BITS {
-            let (hi, lo) = big_digit::from_doublebigdigit(d);
-            data.push(lo);
-            d = hi as DoubleBigDigit;
+            data.push(d);
             dbits -= big_digit::BITS;
+            d = (c as BigDigit) >> (bits - dbits);
```

Comment:
At first I didn't see how the `<< dbits` above could work without overflowing, but then I figured out that you're recovering those lost bits here.  This deserves some quick comments to explain.